packages/flutter/test/widgets/parent_data_test.dart

```diff
@@ -250,7 +251,136 @@ void main() {
+    final List<Object> exceptions = <Object>[];
+    final FlutterExceptionHandler? oldHandler = FlutterError.onError;
+    FlutterError.onError = (FlutterErrorDetails details) {
+      exceptions.add(details.exception);
+    };
+
+    await tester.pumpWidget(
+      const Directionality(
+        textDirection: TextDirection.ltr,
+        child: Stack(
+          textDirection: TextDirection.ltr,
+          children: <Widget>[
+            CustomPositionedWidget(
+              bottom: 8.0,
+              child: Positioned(
+                top: 6.0,
+                left: 7.0,
+                child: DecoratedBox(decoration: kBoxDecorationB),
+              ),
+            ),
+          ],
+        ),
+      ),
+    );
+
+    FlutterError.onError = oldHandler;
+    expect(exceptions.length, 2);
+    expect(
+      exceptions[0].toString(),
+      startsWith(
+        'Incorrect use of ParentDataWidget.\n'
+        'Competing ParentDataWidgets are providing parent data to the same RenderObject:\n'
+        '- Positioned(left: 7.0, top: 6.0) (typically placed directly inside a Stack widget)\n'
+        '- CustomPositionedWidget (typically placed directly inside a Stack widget)\n'
+        'A RenderObject can receive parent data from multiple ParentDataWidgets, but they must '
+        'all be of different types.\n'
+        'Usually, this indicates that one or more of the offending ParentDataWidgets listed '
+        "above isn't placed inside a dedicated compatible ancestor widget that it isn't "
+        'sharing with another ParentDataWidget of the same type.\n'
+        'The ownership chain for the RenderObject that received the parent data was:\n'
+        '  DecoratedBox ← Positioned ← CustomPositionedWidget ← Stack ← Directionality ← ', // End of chain omitted, not relevant for test.
+      ),
+    );
+    expect(
+      exceptions[1].toString(),
+      startsWith(
+        'Incorrect use of ParentDataWidget.\n'
+        'Multiple ParentDataWidgets are providing competing ParentData to the same RenderObject:\n'
+        '- Positioned(left: 7.0, top: 6.0) which writes ParentData of type StackParentData\n'
+        '- CustomPositionedWidget which writes ParentData of type StackParentData\n'
+        'A RenderObject can receive parent data from multiple ParentDataWidgets, but the Type of '
+        'ParentData must be unique to prevent one overwriting another.\n'
+        'This can be resolved by using mixins to separate aspects of a '
+        'ParentData class into separate types, mixing them all in on the '
+        'full ParentData Object, such as KeepAlive does with '
+        'KeepAliveParentDataMixin.\n'
+        'The ownership chain for the RenderObject that received the parent data was:\n'
+        '  DecoratedBox ← Positioned ← CustomPositionedWidget ← Stack ← Directionality ← ', // End of chain omitted, not relevant for test.
+      ),
+    );
```

Comment:
(here and elsewhere) I am surprised that we get two separate exceptions for this. Can we only throw one that contains all the relevant information? (I think you could pretty much achieve this by folding both exceptions into one).

With the two exceptions it feels like we are spamming people's and make them do the work of hunting down the right information from both.